### DISCUSSION TITLE
show FPS yolov5

### DISCUSSION THREAD

**Discussion by HuumbleBee** [ultralytics/yolov5#6713] [Q&A] [answered]
Hello i want to show fps yolov5 object detection on cv2, i have search how to show it, but i still not success to do it. can anyone can direct me where can i put fps computing program so that if i running detect.py fps can appear in cv2? thank you

**Comment by glenn-jocher** [comment, marked as answer]
@HuumbleBee just divide 1000 ms by any of these speeds to get FPS:
<img width="1022" alt="Screenshot 2022-02-21 at 14 50 30" src="https://user-images.githubusercontent.com/26833433/154967934-ec4dc2ec-4327-4e0e-8b1b-447b54cf5a6f.png">

**Reply by HuumbleBee** [reply]
sorry, can you give some example sir? thanks

**Reply by HuumbleBee** [reply]
where i should place devide program for fps sir? detect.py or other file?

**Reply by hoseinhamzei** [reply]
that is my question too how to print average frames per second too? could anyone solve this problem? when i run detect.py on a video i want fps logged too I tried a lot but couldn't get the real fps
update:
with this code I managed to convert inference time to fps when running, it prints 22fps for custom trained yolov5L and about 80 for default yolov5s is it correct? how can I get the final average frame per second
`        
# detect.py line 199
        # Print time (inference-only)
        LOGGER.info(f'{s}Done. ({t3 - t2:.3f}s)')
        # convert inference time in milliseconds to frames per second as well?
        fpsm = 1 / (t3 - t2)
        LOGGER.info(f'FPS: {fpsm:.1f}')`

**Reply by glenn-jocher** [reply]
just divide 1000 ms by any of these speeds to get FPS:
<img width="1022" alt="Screenshot 2022-02-21 at 14 50 30" src="https://user-images.githubusercontent.com/26833433/154967934-ec4dc2ec-4327-4e0e-8b1b-447b54cf5a6f.png">

**Reply by hoseinhamzei** [reply]
`Speed: 0.7ms pre-process, 35.6ms inference, 2.0ms NMS per image at shape (1, 3, 640, 640)`
thanks but sorry i don't understand, here are three values (speed, inference, NMS per image) which one should i divide 1000 by to get average FPS when running the detection on a video?

**Reply by glenn-jocher** [reply]
whichever you want based on your definition of FPS

**Reply by phuctrang** [reply]
in my project, i get 1000/(pre-process+inference+NMS)

**Reply by zehranrgi** [reply]
For example, 
the time taken for pre-processing is 0.8ms, the time taken for inference is 13.2ms, and the time taken for NMS is 3.4ms. Therefore, the total time taken to process one image is 17.4ms.

To calculate the FPS, you need to divide the number of milliseconds in one second (1000) by the time taken to process one image in milliseconds:

FPS = 1000 / 17.4 = 57.47

So, the FPS for this YOLOv5 model on this particular image is approximately 57.47.

**Comment by redhair-vibing** [comment]
> Hello i want to show fps yolov5 object detection on cv2, i have search how to show it, but i still not success to do it. can anyone can direct me where can i put fps computing program so that if i running detect.py fps can appear in cv2? thank you

have you solved your question? I also want to know how

**Reply by michaelkedev** [reply]
You can use ```cv2.putText()```

**Comment by redhair-vibing** [comment]
my code is :

`#!/usr/bin/python3
import cv2
import torch
from PIL import Image
import time

cap = cv2.VideoCapture(0)
print("capture device is open: " + str(cap.isOpened()))
fps_start_time = time.time()
fps=0

while True:

    # img = infence(frame)
    # caculate fps
    success, frame = cap.read()
    fps_end_time = time.time()
    fps_diff_time = fps_end_time - fps_start_time
    fps = 1 / fps_diff_time
    fps_start_time = fps_end_time
    fps_text="FPS:{:.2f}".format(fps)
    cv2.putText(frame, fps_text, (5, 30), cv2.FONT_HERSHEY_COMPLEX, 1, (0, 255, 255), 1)
    cv2.imshow("webcam",frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break
cap.release()

`

**Reply by AbrarKhan009** [reply]
Hello everyone, i am sharing a youtube video where he guide you how to show FPS on real time running video on YOLOv7  maybe it will help you and you can modify the code for yolov5 also after that.

https://www.youtube.com/watch?v=_CkXDjmT8dc&ab_channel=TheCodingBug

**Comment by supriamir** [comment]
Hi @glenn-jocher,

I have calculate the FPS by 1000/speeds , where speed sum of (pre-processing, inference time, post-processing).
'Im encountering confusion with my FPS calculation results. Despite having a model with fewer parameters, it's exhibiting a higher FPS. What could be the reason for this?

![fps](https://github.com/ultralytics/yolov5/assets/55084365/f99a0da1-cf06-48b7-b29d-a023662209b5)
